### ISSUE TITLE
hw06 ready for grading

### ISSUE THREAD

**Issue by menglinzhou** [menglinzhou/STAT545-hw-Zhou-Menglin#6]
@vincenzocoia @gvdr @ksedivyhaley @JoeyBernhardt @mynamedaike @pgonzaleze @derekcho
Thank you!

**Comment by gbraich** [comment]
Hello!

Great job on tackling the assignment.  Part 1 (exercises from Strings) was well done, it looks as though you tackled a few problems from each of the 13 exercises. It was difficult to view some of the answers in the md file, especially html widgets like str_view.  These would be easier to view in an html file.  In the future you might want to use https://rawgit.com to input your html file to resolve these issues.

For exercise 14.3.3.1, regarding British vs American English something you could try that I found to work would be the following regex: str_view(words, "..our$", match=TRUE).  This would find British words that have an extra u (eg. labour, colour etc.). Another tip is that you could use [^ ]+ instead of ([A-Za-z]+) as a regex for any word.

The writing functions section appears to be done correctly, but it was a bit hard to follow without additional commentary.  There was also a png after the experimental function that was not viewable.  Nice job on picking with problem, writing functions is a challenging task!

**Comment by suminwei2772** [comment]
Hi @menglinzhou !
Overall, good job and very thorough job completing all the exercises in the `Character data` task! Be careful with spelling as there were many spelling errors in your explanations and responses.

1. Character data:
a) Good job exploring `paste()`, `paste0()` and `str_c()`. You were thorough in explaining what the arguments `sep`, `collapse` and how `NA` is treated in different functions.
b) Clever use of `str_length()` and `str_sub()` to extract the middle character for strings with even and odd number of characters, by diving n by the appropriate number.
c) Good job writing the function to turn a vector into a string - I liked all your checks in your function to see if the string contained `NA`, or `null`.
d) I learned a lot by going through your responses on regular expressions. For example - (1) using `str_view()` to find words that start with a consonant by using the `[aeious]` and `match=FALSE` arguments, (2) Using `[^c]ie` to check the rule `i` before ` except after `c`, and (3) using {n,} to get words that have three or more of the expression you are looking for. 

2. Writing functions
a) Your quadratic lm function is quite advanced in its design and formulation. Well done! I wasn't sure when you input pop and gdpPercap together into `qlm` you were essentially supplying 2 dependent variables/covariates?
b)  It would have been nice if you had shown your plots generated from output of your experimental functions. I am not familiar with local polynomial estimation. It would've been good also, if you showed what the formulae for your fits looked like, just so we get a more visual/concrete idea of what your function is actually doing. Since your `f` function is quite abstract and difficult to understand from first review, some comments explaining each line of your `f` function code would have been helpful. For instance, what is `solve(t(z)%*%w%*%z)%*%t(z)%*%w%*%y` supposed to do?

Also, don't forget to report your process and let others know the problems/difficulties you encountered. I noticed this section in your response is missing!

**Comment by ksedivyhaley** [comment]
**At least two topics:** Yes (character data, functions)
**Character data (13 sections):** Yes
**Functions:** OK
**Reflection:** Yes 

**Comments:**

- You've completed most of the character data questions, and at least one from each section. The responses are clear.
- For the “flickered ” question, it's better practice to specify word boundaries rather than spaces in case you have colours that don't have a space in front (eg: “The sky in the west is tinged with orange-red.”)
- The function could use a bit more explanation, a more descriptive name, and some basic error checking to make sure the input is reasonable. It looks like it gets the job done, though.

*Your mark will be distributed later. If you would like more feedback, please feel free to message me on slack.*
